src/allmydata/test/test_introducer.py

```diff
@@ -314,6 +316,64 @@ def test_duplicate(self):
 
 NICKNAME = u"n\u00EDickname-%s" # LATIN SMALL LETTER I WITH ACUTE
 
+def foolscapEndpointForPortNumber(portnum):
+    """
+    Create an endpoint that can be passed to ``Tub.listen``.
+
+    :param portnum: Either an integer port number indicating which TCP/IPv4
+        port number the endpoint should bind or ``None`` to automatically
+        allocate such a port number.
+
+    :return: A two-tuple of the integer port number allocated and a
+        Foolscap-compatible endpoint object.
+    """
+    if portnum is None:
+        from twisted.internet import reactor
+        from twisted.internet.interfaces import IReactorSocket
+        if IReactorSocket.providedBy(reactor):
+            # On POSIX we can take this very safe approach of binding the
+            # actual socket to an address.  Once the bind succeeds here, we're
+            # no longer subject to any future EADDRINUSE problems.
+            import fcntl
+            s = socket()
```

Comment:
import to top? Or is the IReactorSocket part telling us "we're on POSIX so the import can only succeed there"..?